CorsixTH/Lua/map.lua

```diff
@@ -660,49 +660,50 @@ function Map:draw(canvas, sx, sy, sw, sh, dx, dy)
         break
       elseif screenY > -32 then
         repeat
-          if screenX < -32 then -- luacheck: ignore 542
-          elseif screenX < sw + 32 then
-            local xy = y * self.width + x
-            local xpos = dx + screenX - 32
-            local ypos = dy + screenY
-            if self.debug_flags then
-              local flags = self.debug_flags[xy]
-              if flags.passable then
-                self.cell_outline:draw(canvas, 3, xpos, ypos)
-              end
-              if flags.hospital then
-                self.cell_outline:draw(canvas, 8, xpos, ypos)
-              end
-              if flags.buildable then
-                self.cell_outline:draw(canvas, 9, xpos, ypos)
-              end
-              if flags.travelNorth and self.debug_flags[xy - self.width].passable then
-                self.cell_outline:draw(canvas, 4, xpos, ypos)
-              end
-              if flags.travelEast and self.debug_flags[xy + 1].passable then
-                self.cell_outline:draw(canvas, 5, xpos, ypos)
-              end
-              if flags.travelSouth and self.debug_flags[xy + self.width].passable then
-                self.cell_outline:draw(canvas, 6, xpos, ypos)
-              end
-              if flags.travelWest and self.debug_flags[xy - 1].passable then
-                self.cell_outline:draw(canvas, 7, xpos, ypos)
-              end
-              if flags.thob ~= 0 then
-                self.debug_font:draw(canvas, "T"..flags.thob, xpos, ypos, 64, 16)
-              end
-              if flags.roomId ~= 0 then
-                self.debug_font:draw(canvas, "R"..flags.roomId, xpos, ypos + 16, 64, 16)
+          if screenX >= -32 then
```

Comment:
How can we clean this up. I spent a little time looking at I believe to make it really clean you need a scanline iterator like the one in c++ that iterates over tiles diagonally.

what you wrote is:
```
if tile_is_past_left_edge_of_screen then
  if tile_is_before_right_edge_of_screen then
    draw_debug_tile
  else
    break
  end
end
```

It could be a little more manageable if the actual draw_debug_tile logic was moved to a separate function. I suggest doing that, then you aren't indenting nearly as much code.